src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;

```

Comment:
Well, at very first sight, no. The session (or Rserve fork process) I start via (RSessionWrapper.open()), is open on demand, then closed when the task is done. Since we are using some kind of TCP/IP protocol. I do not want to compromise the entire application by instantiating a CentWave dedicated Rserve session once and for all as we used to do with JRI. That is not the policy ;)... The strategy is to keep the app working even if something bad happen during a computation (Rserve session crashed, eval failed, connection failed, parallel tasks are taking all CPU available...). If the CentWave deconvolution that just ran crashed/blocked for whatever reason, I want to be burn proof and able to run another deconvolution right after the failure, having MZmine still behaving well in the meanwhile.

I'll have a closer new fresh look at MZmine's model (it's been a while I haven't), but I am pretty sure we can't do anything even tricky one from the resolver's constructor...